optuna/storages/grpc/_server.py

```diff
@@ -0,0 +1,484 @@
+from __future__ import annotations
+
+from concurrent.futures import ThreadPoolExecutor
+import copy
+from datetime import datetime
+import json
+import threading
+
+from optuna.distributions import distribution_to_json
+from optuna.distributions import json_to_distribution
+from optuna.exceptions import DuplicatedStudyError
+from optuna.storages import RDBStorage
+from optuna.storages.grpc import _api_pb2
+from optuna.storages.grpc import _api_pb2_grpc
+from optuna.storages.grpc._grpc_imports import _imports
+from optuna.study._study_direction import StudyDirection
+from optuna.trial._frozen import FrozenTrial
+from optuna.trial._state import TrialState
+
+
+if _imports.is_successful():
+    from optuna.storages.grpc._grpc_imports import grpc
+
+
+DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
+
+
+class OptunaStorageProxyService(_api_pb2_grpc.StorageServiceServicer):
+    def __init__(self, storage_url: str) -> None:
+        self._backend = RDBStorage(storage_url)
```

Comment:
The `storage_url` in Optuna is not exclusive to `RDBStorage`. It can also be used to specify JournalStorage with either JournalFileBackend or JournalRedisBackend.
https://github.com/optuna/optuna/blob/3aeebbcdd233a335c31d0ad68725e786cf8f57ef/optuna/cli.py#L57-L75

Is there something concern that OptunaStorageProxyService should be restricted to RDBStorage only?

Reply:
> Is there something concern that OptunaStorageProxyService should be restricted to RDBStorage only?

Yes, it is. Currently, the server side implementation uses the RDBStorage specific methods such as `RDBStorage._create_new_trial` and `RDBStorage._get_trials`. This is because I would like to introduce the same cache mechanism as `_CachedStorage`. What do you think?

Reply:
I fixed the server only accept the instance of RDBStorage. Thanks for the comment at the mob review.